pennylane_qiskit/converter.py

```diff
@@ -561,36 +543,242 @@ def load_qasm_from_file(file: str):
+    if not isinstance(condition[0], expr.Expr):
+        # Prepare the classical bits used for the condition
+        clbits = [condition[0]] if isinstance(condition[0], Clbit) else list(condition[0])
+
+        # Proceed only if we have access to all conditioned classical bits
+        meas_pl_op = None
+        if all(clbit in mid_circ_regs for clbit in clbits):
+            # Build an integer representation for each switch case
+            meas_pl_op = sum(2**idx * mid_circ_regs[clbit] for idx, clbit in enumerate(clbits))
+
+    # if the target is an Expr
+    else:
+        meas_pl_op = _expr_evaluation(condition[0], mid_circ_regs)
+
+    meas_pl_ops = []
+    if meas_pl_op is not None:
+        # Add a measurement for each of the switch cases
+        meas_pl_ops.extend([meas_pl_op == clval for clval in condition[1]])
+        # If we have default case, add an additional measurement for it
+        if condition[2] == "SwitchDefault":
+            meas_pl_ops.append(
+                reduce(
+                    lambda m0, m1: m0 & m1,
+                    [(meas_pl_op != clval) for clval in condition[1]],
+                )
+            )
+    return meas_pl_ops
+
+
+# pylint:disable = unbalanced-tuple-unpacking
+def _expr_evaluation(condition, mid_circ_regs):
+    """Evaluates the expr condition"""
+
+    # Maps qiskit `expr` names to their mathematical logic
+    _expr_mapping = {
+        "BIT_AND": lambda meas1, meas2: meas1 & meas2,
+        "BIT_OR": lambda meas1, meas2: meas1 | meas2,
+        "BIT_XOR": lambda meas1, meas2: meas1 ^ meas2,
+        "LOGIC_AND": lambda meas1, meas2: meas1 and meas2,
+        "LOGIC_OR": lambda meas1, meas2: meas1 or meas2,
+        "EQUAL": lambda meas1, meas2: meas1 == meas2,
+        "NOT_EQUAL": lambda meas1, meas2: meas1 != meas2,
+        "LESS": lambda meas1, meas2: meas1 < meas2,
+        "LESS_EQUAL": lambda meas1, meas2: meas1 <= meas2,
+        "GREATER": lambda meas1, meas2: meas1 > meas2,
+        "GREATER_EQUAL": lambda meas1, meas2: meas1 >= meas2,
+    }
+
+    # Get the left and right classical controls
+    cond1, cond2 = condition.left, condition.right
+
+    # Iterate over each and extract classical bits
+    clbits, clvals = [], []
+    for _, carg in enumerate([cond1, cond2]):
+        # We don't need to work with Qiskit's Cast expr op,
+        # as we'll be casting stuff manually by ourselves.
+        if isinstance(carg, expr.Cast):
+            carg = carg.operand
+
+        if isinstance(carg, expr.Value):
+            clvals.append([carg.value])
+        elif isinstance(carg, expr.Var):
+            clbits.append([carg.var] if isinstance(carg.var, Clbit) else list(carg.var))
+
+    # Proceed only if we have access to all conditioned classical bits
+    for idx, clreg in enumerate(clbits):
+        if not all(clbit in mid_circ_regs for clbit in clreg):
+            return None
+        clbits[idx] = [mid_circ_regs[clbit] for clbit in clreg]
```

Comment:
are we supposed to be overriding values here?

Reply:
Yes, this is intentional. What it does is map Qiskit's classical bits for each register to the corresponding mid-circuit PL measurement that we keep track of in the `mid_circ_regs`.